### ISSUE TITLE
IPTT report: Pin this IPTT

### ISSUE THREAD

**Issue by axoplasm** [mercycorps/TolaActivity#588]

**Comment by axoplasm** [comment]
I'm not sure what the backend issues are for this so I made it "high effort." we might want to separate the UI/Frontend as a separate task for estimation purposes

**Comment by KennyMonster** [comment]
Would it make sense to also show a list of pinned reports for the selected program on the IPTT report filter bar?

Also, should there a way to delete a pinned report? I could imagine a user wanting to get rid of one if there's a mistake in the filter criteria.

**Comment by sanjuroj** [comment]
I'd say "Yes" to both.  @jennyhavah?

**Comment by jennyhavah** [comment]
@KennyMonster and @sanjuroj. Regarding pinning to the IPTT view itself, I hadn't thought of that. It's an interesting idea! But I have concerns about the complexity it would add to an already complex panel. I am not convinced it would add enough value to justify the complexity. For example, you'd have to keep changing the list of reports depending on what program the IPTT is being run on. And what happens when we have multi-program IPTTs, which we may introduce in the future? 

So for now, let's not tackle this. Not ruling it out, but I think it needs more exploration before diving in.

**Comment by KennyMonster** [comment]
> 2.1. The popover is dismissable by clicking the tab or escape key, or by clicking anywhere outside the popover.

Any objection to not closing on the `tab` key? The reason being is tab could potentially be used to tab through the form + button elements while doing the data entry.

**Comment by jennyhavah** [comment]
@KennyMonster Good point! What is standard behavior in your experience? For example, would it be expected behavior to tab through the text field and button and then the next tab after the button is to close? 

Just want to make sure we give people a way to close the popover with the keyboard, for people who can't/don't want to use the mouse for whatever reason.

**Comment by KennyMonster** [comment]
@jennyhavah I think Esc is pretty universal for closing modals/popovers (it really annoys me when this doesn't work!), while Tab is instead used solely for redirecting focus to interactive page elements such and buttons, links, and input fields. In this context, the popover would only close via Tab if you tab to a close button (such as "X" in the corner) and press Enter to activate it.

My understanding is the focus cycling behavior is important in certain accessibility situations where a mouse like device isn't easily available, or a user is using some sort of alternative input device which generates keystrokes such as voice recognition.

**Comment by esinn** [comment]
This might be out of scope or a stretch but what about when you try to pin a report with the same criteria that you've already pinned? Ideally it would be awesome if it would say, you've already pinned this report or something of that nature. @jennyhavah thoughts? Maybe that's a big LOE.

**Comment by jennyhavah** [comment]
@esinn I can see how such a reminder would come in handy. However, I would rate it relatively low priority for the following reasons:
1. This type of duplication seems unlikely to occur often. (That's my assumption -- we might hear otherwise from the field.)
2. If duplication does occur, it doesn't cause much harm -- two links going to the same place -- and it only affects you, the creator of the pin.
3. The duplication is easily fixed by deleting either of the duplicate pins.

Now, if the pins were actual "objects" (like our Documents) and were shared across accounts, that would be a whole other story and I'd be very much in favor of preventing duplication. We'll be tackling an issue of this very nature in Huckleberry!

**Comment by esinn** [comment]
Got it. Then this ticket passes all criteria.
